libs/remote_storage/src/s3_bucket.rs

```diff
@@ -75,6 +79,14 @@ impl S3Bucket {
             )
             // uses imds v2
             .or_else("imds", ImdsCredentialsProvider::builder().build())
+            // uses "AWS_WEB_IDENTITY_TOKEN_FILE", "AWS_ROLE_ARN", "AWS_ROLE_SESSION_NAME"
+            // needed to access remote extensions bucket
+            .or_else(
```

Comment:
Do you think, we really need to optimize this?

s3 client is created only once for extensions downloader,
and for pageserver codepath nothing changed

Alternatively we can add a flag `S3Bucket::new(use_token_provider: bool)` to distinguish two use cases.

Reply:
Agreed pageserver does not need it, but the imds queries are a rate-limited resource as far as I know, and this would be another hinderance on compute startup times if we were to do always do an imds credential lookup *before* checking out the 3 env variables.

We cannot impose a ratelimit on multiple computes starting up on the same host because the computes cannot communicate to each other, so it might happen that we'd see an increased error rate with busy nodes..

Checking the imds...

Reply:
> Checking the imds...

So yeah, it would appear that an http request would be made: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-metadata-v2-how-it-works.html